src/networking/ICommunicator.java

```diff
@@ -0,0 +1,19 @@
+package networking;
+/**
+ *
+ * @author Prudhvi Vardhan Reddy Pulagam
+ */
+public interface ICommunicator {
+
+    //Start communication by initalizing all the threads needed for communication. 
+    void start();
+    
+    //Stop the Communication.
+    void stop();
+    
+    //Send the data to the destination host.
+    void send(String destination, String message, String identifier);
+    
+    //Subscribe the module for receiving the data from the network module.
+    void subscribeForNotifications(String identifier, INotificationHandler handler);
+}
```

Comment:
Please use block tags and multiline comments for describing methods. We need to keep it consistent with the code base.